### DISCUSSION TITLE
Add ability to disable CSRF

### DISCUSSION THREAD

**Discussion by ajsharp** [nextauthjs/next-auth#4992] [Ideas]
### Description 📓

Currently it's not possible to use next-auth without a CSRF token. This makes it impossible to use your next-auth powered API outside a next app. CSRF token's are only used for web app requests bc their point is to protect against XSS attacks.

### How to reproduce ☕️

Try to submit an api request without an CSRF token.

### Contributing 🙌🏽

Yes, I am willing to help implement this feature in a PR

**Comment by andrewgbliss** [comment]
Has this been implemented yet?

**Comment by MasterLambaster** [comment]
For anyone who'd came across this question
https://github.com/nextauthjs/next-auth/blob/8b38d324309796b95cb1ce515be77d2922f9d49d/packages/core/src/lib/index.ts#L190

**Reply by henricook** [reply]
@MasterLambaster Any chance you could include an example of how to use this?

**Reply by MasterLambaster** [reply]
@henricook  It's an option that you have to pass to the `NextAuth` alongside with providers and other options, smth like 
```javascript
const authConfig = {
  providers: [
    GitHubProvider({
      clientId: process.env.GITHUB_ID,
      clientSecret: process.env.GITHUB_SECRET,
    }),
  ],
  skipCSRFCheck: true,
};

export default async function handler(req, res) {
  const { nextauth, provider, ...rest } = req.query
  req.query = { nextauth: [nextauth, provider], ...rest }
  return await NextAuth(req, res, authConfig)
}
```

**Reply by xjunior** [reply]
Is this released? I'm using next-auth 4.22.1 and skipCSRFCheck is not in it

**Reply by ItzSudhan** [reply]
it's been a year and its still on beta not released yet. It has many breaking changes though

**Comment by vivekfyi** [comment]
any update on this
